### ISSUE TITLE
is there interest in adding option for beginning padding to `ggml_pad`

### ISSUE THREAD

**Issue by balisujohn** [ggerganov/ggml#861]
right now `ggml_pad`only supports putting padding at the end of each dimension. Is there support for generalizing it to allow adding padding at at the beginning of each dimension? My thinking is the arguments would be 
````
p00,p01, p10,p11,p20,p21,p30,p31
````

This is necessary for tortoise.cpp so as usual I'll be implementing it anyway and am happy to upstream it.

**Comment by ggerganov** [comment]
Yes, make sure to extend the existing kerenels and not introduce new ones. The `ggml` API can keep `ggml_pad` and also add `ggml_pad_ext` with the full parameter list
